app/components/modals/startApplicationModal.tsx

```diff
@@ -0,0 +1,148 @@
+"use client";
+import React, { useState } from "react";
+import { Button } from "../ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "../ui/dialog";
+import { useToast } from "../ui/use-toast";
+import axios from "axios";
+import { useSession } from "next-auth/react";
+import { useForm } from "react-hook-form";
+import { AppliedStaffProfileResponse } from "@/app/(staff)/job-posts/[id]/page";
+
+const isApplicationStarted = (
+  appliedStaffProfiles: AppliedStaffProfileResponse[],
+  staffId: string | null
+) => {
+  if (!staffId) {
+    return false;
+  }
+  console.log(appliedStaffProfiles);
+
+  for (let appliedStaffProfile of appliedStaffProfiles) {
+    if (appliedStaffProfile.staffProfile.id === staffId) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+interface StartApplicationModalProps {
+  jobPostId: string;
+  appliedStaffProfiles: AppliedStaffProfileResponse[];
+}
+
+export function StartApplicationModal({
+  jobPostId,
+  appliedStaffProfiles,
+}: StartApplicationModalProps) {
+  const { data: session } = useSession();
+  const [applicationStarted, setApplicationStarted] = useState<boolean>(
+    isApplicationStarted(
+      appliedStaffProfiles,
+      session?.user.staffProfile?.id || null
+    )
+  );
+  const [openModal, setOpenModal] = useState(false);
```

Comment:
because you're controlling your own state for this modal closing the modal using the 'x' icon or by clicking outside the modal is not working